### DISCUSSION TITLE
Simpler & faster way to enable all freemods in multiplayer

### DISCUSSION THREAD

**Discussion by timiimit** [ppy/osu#23560] [Ideas] [answered]
Same as in stable, i think there should be a faster way to enable all mods. In stable, there is a simple checkbox on/off and everyone is used to that. Often when i see people trying out multi in lazer for the first time they are confused about this new system. I think there should be a way to enable only specific mods (which is especially useful for tournaments), but that should be a more advanced option. For most of the cases people either want all or none.

I propose the idea to add one of the following:
- Checkbox (left side is a checkbox, right opens the mod select)
![image](https://github.com/ppy/osu/assets/32331609/d451f750-3e17-4c86-ae80-93db368acac4)

- something like `...` or `^` (left side is a checkbox, right opens the mod select)
![image](https://github.com/ppy/osu/assets/32331609/ece3b624-c6f9-4cd6-a53f-033df1c41f9f)

- a way to "drag the menu" out of the button and make the button itself act as a checkbox

This change will only reduce 1 click, but the animations for selecting mods take long when pressing "select all" so i believe it is worth it. I probably spent 30 min of my life by now waiting for all mods to enable. Seconds add up fast.

**Comment by peppy** [comment]
Very much agree with this.

@arflyte want to provide a design update with this in mind?

**Comment by marioman63** [comment]
![image](https://github.com/ppy/osu/assets/76608775/6cba73c4-b8e0-4aec-9868-5ff91b2cd016)
A change like the one suggested would alleviate a concern of mine. As more mods have been added, enabling all mods pushes the "random" and "options" buttons further to the right. After the latest batch of mods was added, I noticed that the "options" button is partially covered by the logo button in the corner.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/pull/24294
